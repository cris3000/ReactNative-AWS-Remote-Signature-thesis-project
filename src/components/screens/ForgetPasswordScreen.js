// Import React
import React from 'react'
//Import Styling elements from React native
import {
  TouchableOpacity,
  TouchableWithoutFeedback,
  StyleSheet,
  Text,
  SafeAreaView,
  StatusBar,
  KeyboardAvoidingView,
  Keyboard,
  View,
  Alert,
  Animated
} from 'react-native'
// imports Layout elements from native base
import {
  Container,
  Item,
  Input} from 'native-base'

// Import AWS AUTHENTICATION
import Auth from '@aws-amplify/auth'

//Forgot passward Class
export default class ForgetPasswordScreen extends React.Component {
  //State to hold vars
  state = {
    username: '',
    authCode: '',
    newPassword: '',
    fadeIn: new Animated.Value(0),  // Initial value for opacity: 0
    fadeOut: new Animated.Value(1),  // Initial value for opacity: 1
    isHidden: false
  }

  onChangeText(key, value) {
    this.setState({
      [key]: value
    })
  }
  // async Function To Request a users New Password
  async forgotPassword() {
    const { username } = this.state
    await Auth.forgotPassword(username)
    .then(data => console.log('New code sent', data))
    .catch(err => {
      if (! err.message) {
        Alert.alert('Error: ', err)
      } else {
        Alert.alert('Error: ', err.message)
      }
    })
  }
  // async Function to Redirect User to Signin Screen once the request has been made
  async forgotPasswordSubmit() {
    const { username, authCode, newPassword } = this.state
    await Auth.forgotPasswordSubmit(username, authCode, newPassword)
    .then(() => {
      this.props.navigation.navigate('SignIn')
    })
    .catch(err => {
      if (! err.message) {
        Alert.alert('Error: ', err)
      } else {
        Alert.alert('Error: ', err.message)
      }
    })
  }
  //Render Function
  render() {
    
    return (
      <SafeAreaView style={styles.container}>
        <StatusBar/>
        <KeyboardAvoidingView 
          style={styles.container} 
          behavior='padding' 
          enabled 
          keyboardVerticalOffset={23}>
          <TouchableWithoutFeedback style={styles.container} onPress={Keyboard.dismiss}>
            <View style={styles.container}>
   
              <View style={styles.logoContainer}>
             
              </View>
              {/* Information */}
              <Container style={styles.infoContainer}>
                <View style={styles.container}>
                  {/* User detail */}
                  <Item style={styles.itemStyle}>
                  
                    <Input
                      style={styles.input}
                      placeholder='Username'
                      placeholderTextColor='#adb4bc'
                      keyboardType={'email-address'}
                      returnKeyType='go'
                      autoCapitalize='none'
                      autoCorrect={false}
                      onChangeText={value => this.onChangeText('username', value)}
                      onFocus={this.fadeOut.bind(this)}
                      onEndEditing={this.fadeIn.bind(this)}
                    />
                  </Item>
                  <TouchableOpacity
                    onPress={() => this.forgotPassword()}
                    style={styles.buttonStyle}>
                    <Text style={styles.buttonText}>
                      Send Code
                    </Text>
                  </TouchableOpacity>
                  {/* Section that allows new password,*/}
                  <Item style={styles.itemStyle}>
                  
                    <Input
                      style={styles.input}
                      placeholder='New password'
                      placeholderTextColor='#adb4bc'
                      returnKeyType='next'
                      autoCapitalize='none'
                      autoCorrect={false}
                      secureTextEntry={true}
                      onSubmitEditing={(event) => { this.refs.SecondInput._root.focus() }}
                      onChangeText={value => this.onChangeText('newPassword', value)}
                      onFocus={this.fadeOut.bind(this)}
                      onEndEditing={this.fadeIn.bind(this)}
                    />
                  </Item>
                  {/* Section That Confirms the Code Sent two-factor authentication either SMS or Email  */}
                  <Item style={styles.itemStyle}>
                   
                    <Input
                      style={styles.input}
                      placeholder='Confirmation code'
                      placeholderTextColor='#adb4bc'
                      keyboardType={'numeric'}
                      returnKeyType='done'
                      autoCapitalize='none'
                      autoCorrect={false}
                      secureTextEntry={false}
                     
                      onChangeText={value => this.onChangeText('authCode', value)}
                    
                    />
                  </Item>
                  <TouchableOpacity
                    onPress={() => this.forgotPasswordSubmit()}
                    style={styles.buttonStyle}>
                    <Text style={styles.buttonText}>
                      Confirm the new password
                    </Text>
                  </TouchableOpacity>
                </View>
              </Container>
            </View>
          </TouchableWithoutFeedback>
        </KeyboardAvoidingView>
      </SafeAreaView>
    );
  }
}
//Styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#5059ae',
    justifyContent: 'center',
    flexDirection: 'column'
  },
  input: {
    flex: 1,
    fontSize: 17,
    fontWeight: 'bold',
    color: '#fff',
  },
  infoContainer: {
    position: 'relative',
    left: 0,
    right: 0,
    height: 300,
    bottom: 25,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 30,
    backgroundColor: '#5059ae',
  },
  itemStyle: {
    marginBottom: 20,
  },
  iconStyle: {
    color: '#fff',
    fontSize: 28,
    marginRight: 15
  },
  buttonStyle: {
    alignItems: 'center',
    backgroundColor: '#b44666',
    padding: 14,
    marginBottom: 20,
    borderRadius: 3,
  },
  buttonText: {
    fontSize: 18,
    fontWeight: 'bold',
    color: "#fff",
  },
  logoContainer: {
    position: 'absolute',
    left: 0,
    right: 0,
    height: 600,
    bottom: 180,
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
})