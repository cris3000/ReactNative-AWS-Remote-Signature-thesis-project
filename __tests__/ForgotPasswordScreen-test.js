// __tests__/HomeScreen-page-test.js
import 'react-native';
import React from 'react';
import ForgetPasswordScreen from '../src/components/screens/ForgetPasswordScreen';

import renderer from 'react-test-renderer';
// Testing whether screen renders
test('renders correctly', () => {
  const tree = renderer.create(
   <ForgetPasswordScreen />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});