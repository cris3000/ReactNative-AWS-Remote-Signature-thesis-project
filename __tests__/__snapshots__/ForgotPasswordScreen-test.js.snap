// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#5059ae",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#5059ae",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#5059ae",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 180,
            "flex": 1,
            "height": 600,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
              "height": 1334,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#5059ae",
              "bottom": 25,
              "flexDirection": "row",
              "height": 300,
              "justifyContent": "center",
              "left": 0,
              "paddingHorizontal": 30,
              "position": "relative",
              "right": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#5059ae",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginBottom": 20,
                "marginLeft": 2,
                "opacity": 1,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              editable={true}
              keyboardType="email-address"
              onChangeText={[Function]}
              onEndEditing={[Function]}
              onFocus={[Function]}
              placeholder="Username"
              placeholderTextColor="#adb4bc"
              rejectResponderTermination={true}
              returnKeyType="go"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "top": 1.5,
                  },
                  Object {
                    "color": "#fff",
                    "flex": 1,
                    "fontSize": 17,
                    "fontWeight": "bold",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#b44666",
                "borderRadius": 3,
                "marginBottom": 20,
                "opacity": 1,
                "padding": 14,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Send Code
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginBottom": 20,
                "marginLeft": 2,
                "opacity": 1,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              editable={true}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="New password"
              placeholderTextColor="#adb4bc"
              rejectResponderTermination={true}
              returnKeyType="next"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "top": 1.5,
                  },
                  Object {
                    "color": "#fff",
                    "flex": 1,
                    "fontSize": 17,
                    "fontWeight": "bold",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginBottom": 20,
                "marginLeft": 2,
                "opacity": 1,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              editable={true}
              keyboardType="numeric"
              onChangeText={[Function]}
              placeholder="Confirmation code"
              placeholderTextColor="#adb4bc"
              rejectResponderTermination={true}
              returnKeyType="done"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "top": 1.5,
                  },
                  Object {
                    "color": "#fff",
                    "flex": 1,
                    "fontSize": 17,
                    "fontWeight": "bold",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#b44666",
                "borderRadius": 3,
                "marginBottom": 20,
                "opacity": 1,
                "padding": 14,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Confirm the new password
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
